kaldb/src/main/java/com/slack/kaldb/chunk/ReadOnlyChunkImpl.java

```diff
@@ -155,10 +155,9 @@ private KaldbMetadataStoreChangeListener cacheNodeListener() {
 
   private void registerSearchMetadata(String snapshotName)
       throws ExecutionException, InterruptedException, TimeoutException {
-    // for "name" we pass url so that we can run two JVMs on the same host and not have znode path
-    // collision.
     this.searchMetadata =
-        new SearchMetadata(searchContext.toName(), snapshotName, searchContext.toUrl());
+        new SearchMetadata(
+            snapshotName + "_" + searchContext.hostname, snapshotName, searchContext.toUrl());
```

Comment:
I think it's best to update the toName() method in SearchContext or add a new method in SearchMetadata to keep all the name building logic in the same place. This logic is already repeated twice here. 

May be make it static so we can use it in unit tests also.

Reply:
Done.